src/Aspire.Hosting.Azure/Provisioning/Internal/DefaultProvisioningContextProvider.cs

```diff
@@ -97,54 +90,160 @@ private async Task RetrieveAzureProvisioningOptions(JsonObject userSecrets, Canc
-                        new InteractionInput { InputType = InputType.Text, Label = AzureProvisioningStrings.ResourceGroupLabel, Value = GetDefaultResourceGroupName() },
-                    ],
-                    new InputsDialogInteractionOptions
+                var selectedSubscriptionId = await PromptForSubscriptionAsync(cancellationToken).ConfigureAwait(false);
+
+                if (selectedSubscriptionId == null)
+                {
+                    return;
+                }
+
+                var (selectedLocation, selectedResourceGroup) = await PromptForLocationAndResourceGroupAsync(selectedSubscriptionId, cancellationToken).ConfigureAwait(false);
+
+                if (selectedLocation == null)
+                {
+                    return;
+                }
+
+                // Set the options
+                _options.SubscriptionId = selectedSubscriptionId;
+                _options.Location = selectedLocation;
+                _options.ResourceGroup = selectedResourceGroup;
+                _options.AllowResourceGroupCreation = true; // Allow the creation of the resource group if it does not exist.
+
+                var azureSection = userSecrets.Prop("Azure");
+
+                // Persist the parameter value to user secrets so they can be reused in the future
+                azureSection["SubscriptionId"] = _options.SubscriptionId;
+                azureSection["Location"] = _options.Location;
+                azureSection["ResourceGroup"] = _options.ResourceGroup;
+
+                _provisioningOptionsAvailable.SetResult();
+            }
+        }
+    }
+
+    private async Task<string?> PromptForSubscriptionAsync(CancellationToken cancellationToken = default)
+    {
+        // Get the available subscriptions based on the current credential or fallback to text input
+        try
+        {
+            var credential = tokenCredentialProvider.TokenCredential;
+            var armClient = armClientProvider.GetArmClient(credential);
+
+            logger.LogInformation("Fetching available Azure subscriptions...");
+            var availableSubscriptions = await armClient.GetAvailableSubscriptionsAsync(cancellationToken).ConfigureAwait(false);
+
+            if (availableSubscriptions.Count == 0)
+            {
+                throw new InvalidOperationException("No Azure subscriptions are available for the authenticated user.");
+            }
+
+            var subscriptionOptions = availableSubscriptions
+                .Select(s => KeyValuePair.Create(s.SubscriptionId, $"{s.DisplayName} ({s.SubscriptionId})"))
+                .ToList();
+
+            var result = await interactionService.PromptInputsAsync(
+                AzureProvisioningStrings.SubscriptionInputsTitle,
+                AzureProvisioningStrings.SubscriptionInputsMessage,
+                [
+                    new InteractionInput
+                    {
+                        InputType = InputType.Choice,
```

Comment:
@JamesNK This selection input revealed a few quirks with the UI for selection.

It doesn't appear to handle inputs that resize the modal well:

<img width="2132" height="1030" alt="Image" src="https://github.com/user-attachments/assets/fd2d69bb-1a9d-47c1-bf9d-dddcc1a91f9e" />

Also, we need some way to present a loading UX in the modal or somewhere as we wait to resolve the values that we populate the choice with.

Reply:
Should be easy to fix with a max width on the dropdown.

Reply:
I couldn't repo this. Long name in select:

<img width="718" height="816" alt="image" src="https://github.com/user-attachments/assets/c7223b7b-a481-4fa7-8660-b5345047cc72" />

What browser are you using?